### DISCUSSION TITLE
Why Databaseless?

### DISCUSSION THREAD

**Discussion by stevebrainng** [realrashid/plan-craft#11] [Q&A]
I was wondering why you gone the database-less route, Because for dynamic usage in a scenario like allowing our client's easily manage subscription plans and features management, I think Database queries can still be optimized to perform required action.

**Comment by realrashid** [comment]
**Why Databaseless with PlanCraft?**

While traditional methods often rely on database queries for managing dynamic subscription plans and features, **PlanCraft** takes a different approach to streamline the process. By eliminating the need for complex database setups, **PlanCraft** enhances efficiency, flexibility, and scalability. While optimized database queries are certainly viable, the database-less route offers distinct advantages. It simplifies development, reduces overhead, and enhances security. With **PlanCraft**, subscription management becomes faster and more agile, empowering clients with seamless control over their subscriptions in Laravel SaaS applications.

**Comment by stevebrainng** [comment]
Okay which means we always have to get into the code to manage the plans, features, etc.

**Comment by realrashid** [comment]
Changing plans isn't a daily task—it's typically set up before launch and adjusted occasionally. 
With **PlanCraft**, managing plans directly in the `PlanCraftServiceProvider` offers a straightforward solution without the need for constant code changes.

**PlanCraft** simplifies plan setup and updates by defining plans directly in the `PlanCraftServiceProvider`. 
This means you can set up plans once and make occasional adjustments easily.
